auth/serviceaccount.go

```diff
@@ -0,0 +1,60 @@
+package auth
+
+import (
+	"context"
+	"fmt"
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/kelseyhightower/envconfig"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+type ServiceAccount struct {
+	UserId string
+}
+
+type ServiceAccountAuthenticator struct {
+	TokenEndpoint string `envconfig:"TIDEPOOL_AUTH_SERVICE_TOKEN_ENDPOINT"`
+}
+
+func NewServiceAccountAuthenticator() (*ServiceAccountAuthenticator, error) {
```

Comment:
Have you considered pulling envconfig out of this function? It would be easier to test, and for other things to use, if the arguments it needed were passed to it, as opposed to needing to be mocked via envconfig.

As a reader of the code, I'd prefer not to have to know anything about envconfig in order to know how the properties of ServiceAccountAuthenticator are being established.

Reply:
The purpose of this function is to instantiate the struct with the configuration from the env. If you want to instantiate it without knowing anything about envconfig you can always do:
```golang
authenticator := &ServiceAccountAuthenticator{
  TokenEndpoint: "https://...."
}
```

Reply:
True. I still think it's a strange constructor, and not at all what I would expect. I'd expect a `New` to do a basic construction without envconfig, and a `NewFromEnvconfig` to do what you've done here. But since there's no other real initialization of maps or channels going on, this is fine.